projects/workers/src/worker/pipes/worker.pipe.ts

```diff
@@ -0,0 +1,26 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {Observable} from 'rxjs';
+import {WebWorker} from '../classes/web-worker';
+import {WorkerExecutor} from '../services/worker-executor.service';
+import {WorkerFunction} from '../types/worker-function';
+
+@Pipe({
+    name: 'waWorker',
+})
+export class WorkerPipe implements PipeTransform {
+    private workers = new WeakMap<WorkerFunction, WebWorker>();
+
+    constructor(private workerExecutor: WorkerExecutor) {}
+
+    transform<T, R>(value: T, fn: WorkerFunction<T, R>): Observable<R> {
+        const worker = this.workers.has(fn)
```

Comment:
= this.workers.get(fn) || this.workerExecutor.createWorker(fn);

Reply:
done